pyteal/compiler/scratchslots_test.py

```diff
@@ -386,7 +386,7 @@ def sub3Impl(a1, a2, a3):
 
     # mainSlot2 and subroutine2Slot1 request the same ID, 100
     with pytest.raises(TealInternalError):
-        actual = assignScratchSlotsToSubroutines(subroutineMapping, subroutineBlocks)
+        actual, _ = assignScratchSlotsToSubroutines(subroutineMapping, subroutineBlocks)
```

Comment:
@algoidurovic I didn't closely review the existing tests, but the diff suggests reserved IDs aren't tested (usage of `_`).  Can you modify/create a test to help build confidence the functionality works as intended?

Reply:
I think this is no longer a concern with the recent changes I've made. Resolving for now but feel free to unresolve.